### PR TITLE
planner/cascades: fill group stats

### PR DESCRIPTION
**Pull request by jackwener**
### What problem does this PR solve?

Issue Number: part of #13709

The statistics are in `group`, but the Cascade Planner did not fill all the Group Stats in the Logical phase.

Both `StreamAgg` and `IndexLookup` need it.

It blocks #14857, which is part of #14831, and most code is from it.

Problem Summary:

### What is changed and how it works?

What's Changed:

fill group stats in `implGroup`

How it Works:

fill group stats

### Check List 

Tests 

- Unit test

### Release Note

None

### CODE REVIEW

**Comment by jackwener** [comment]
PTAL @chrysan

File: planner/cascades/optimize.go [hunk 1]
```diff
@@ -244,6 +251,10 @@ func (opt *Optimizer) onPhaseImplementation(sctx sessionctx.Context, g *memo.Gro
 	prop := &property.PhysicalProperty{
 		ExpectedCnt: math.MaxFloat64,
 	}
+	err := opt.fillGroupStats(g)
```
Comment (Reminiscent) [inline planner/cascades/optimize.go:254]: Should we remove the `fillGroupStats` in `Optimizer.implGroup`?

**Pull request closed without merging**